### PR TITLE
Fix/cal-5557: added keyboard switching view

### PR DESCRIPTION
**Pull request by amandadupell**
#### What does this PR do?

Improved the keyboard to keep active date in view when using the arrow keys

#### Where should the reviewer start?

Calendar.js

#### What testing has been done on this PR?

Tested across multiple Calendar and DateInput stories

#### How should this be manually tested?

Test across Calendar and DateInput stories

#### What are the relevant issues?

#5557 

#### Video of implemented behavior

https://user-images.githubusercontent.com/38971752/132919959-4b2f9f3a-f017-49f3-b012-bc032c8b7297.mov

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

No

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible

### CODE REVIEW

**Review by taysea — COMMENTED** [review]
Overall the behavior looks good. The only thing that was a little strange in the experience was when a date from the next month is visible in my current calendar (but greyed out), when I arrow to it, it already scrolls the calendar to make the next month active. This had the effect of me thinking I might've pressed the down arrow key twice on accident.

My thought is that if a date is visible in the current view, even if it is from the next month, we shouldn't change the active month. This should only happen when the arrow key is moving the active date out of view.

**Comment by amandadupell** [comment]
updated behavior:

https://user-images.githubusercontent.com/38971752/133688375-a2e11aa4-13c3-4ead-b15d-a36153f0902e.mov

File: src/js/components/Calendar/Calendar.js [hunk 1]
```diff
@@ -902,14 +902,30 @@ const Calendar = forwardRef(
               event.preventDefault();
               event.stopPropagation(); // so the page doesn't scroll
               setActive(addDays(active, -7));
+              if (!betweenDates(addDays(active, -7), displayBounds)) {
```
Comment (taysea) [inline src/js/components/Calendar/Calendar.js:905]: nice!

**Comment by taysea** [comment]
For some reason in some cases it's not going up to the next month in this case

https://user-images.githubusercontent.com/12522275/133689228-1e0f3efc-58cc-402d-8b75-6df1c18f7257.mov

**Review by taysea — APPROVED** [review]
LGTM! Good job.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**